### DISCUSSION TITLE
Git completion is slower when run interactively

### DISCUSSION THREAD

**Discussion by Supremist** [fish-shell/fish-shell#10556] [Q&A]
fish, version 3.7.1
Windows 11, msys2
I experience slow git completion. 1-3 seconds, depending on the command and the repository size. Pretty similar to this [issue](https://github.com/fish-shell/fish-shell/issues/8266)
I can see that git completions are written not the most optimal way and together with the msys2 overhead it causes slowness. 
But during the investigation I found something that I can't explain: 
`$ time fish -c 'complete -C"git add t"'`
runs much faster than
`$ complete -C"git add t"`
![image](https://github.com/fish-shell/fish-shell/assets/12670116/30513e24-696c-4055-a061-d892a7afbe6f)

Can somebody explain what is happening? Maybe this could be a lead to fixing this delays?

**Comment by faho** [comment]
It is possible that something in your config does something to interactive or login shells that causes them to be slower. One possible cause is $PATH being different. All I can tell you is to look at your config and see what's holding it up. Try `fish -i -c 'complete -C"git add t"'` to see if that is slow as well, and then run `fish --profile /tmp/git.prof -i -c 'complete -C"git add t"'` followed by `sort -nk2 /tmp/git.prof` to see which part is slow.

However, we also know that msys performs terribly and there's not been a lot of movement in that area - e.g. https://github.com/msys2/MSYS2-packages/issues/138 is still open, as is https://github.com/msys2/MSYS2-packages/issues/1634. To be quite honest, I do not believe fish on msys is in a good state right now (and the next release apparently won't be ready there because the rust port won't be).

**Comment by Supremist** [comment]
I gathered the logs, as was asked in the issue, maybe they will help: 
>$ fish --profile oneline.txt -c 'complete -C"git add t"'
[oneline.txt](https://github.com/user-attachments/files/15738238/oneline.txt)

>$ fish --profile interactive.txt
> $ complete -C"git add t"
> $ exit
[interactive.txt](https://github.com/user-attachments/files/15738262/interactive.txt)
